dist/common/scripts/scylla_util.py

```diff
@@ -293,6 +294,19 @@ def swap_exists():
     swaps = out('swapon --noheadings --raw')
     return True if swaps != '' else False
 
+def wait_for_lock(lock):
+    procs = []
+    for p in psutil.process_iter([ 'name', 'open_files']):
+        if not p.info['open_files']:
+            continue
+        for o in p.info['open_files']:
+            if o.path == lock:
+                print(f"Lock {lock} has held by process {p.pid} ({p.info['name']})")
```

Comment:
I can understand why automation tool might want to do this kind of things

Not sure scylla installation need to go that this extent
user would see the failure of apt (a failure that most of the user/operator are familiar with), and they would know they need to wait, or check what happen with that process using apt

I do think scylla setup need to automated what's going on ones machine to this level

Reply:
i think the problem is, this is _automation_. so minimum human intervention is desired. if user happens to have `unattended-upgrades` or `cron-apt` installed and enabled, it would be nice if we can try again if a certain step fails due to an intermediate failure.

but if we happen to run into a fatal error, i agree with you that we should fail and report the error back to the admin.

Reply:
those are part of scylla installation/setup, installation should be as much as possible part of pkg requirements, those extra installation which are not could fail in all kind of ways, this is one of them, I think it should be minimalistic, and the admin should be attending to those types of things.

also I would do `apt update` on the behalf of the admin either, he probably did it right before he started the installation.

Reply:
> if user happens to have `unattended-upgrades` or `cron-apt` installed and enabled, it would be nice if we can try again if a certain step fails due to an intermediate failure.

Right, normal Ubuntu installation enables unattended-upgrades by default, and it can breaks scylla_setup (#16537).
On our machine-image we simply disabled unattended-upgrades, but we do not want to modify customers environment settings unless it's really needed.
We also don't want to add "Conflicts: unattended-upgrades" on debian/control because of same reason (it might also increase diffculity to install the package).
So we need to find other way to avoid breaking setup.

Reply:
Our own testing can be randomly broken by this problem.